#! /bin/sh /usr/share/dpatch/dpatch-run
## setlocale.dpatch by Samuel Mimram <smimram@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Set locale, see #445193.

@DPATCH@
diff -urNad lablgtk2-2.10.0~/src/gtkMain.ml lablgtk2-2.10.0/src/gtkMain.ml
--- lablgtk2-2.10.0~/src/gtkMain.ml	2007-08-17 03:24:07.000000000 +0000
+++ lablgtk2-2.10.0/src/gtkMain.ml	2007-12-14 15:02:24.000000000 +0000
@@ -43,6 +43,7 @@
       with Error err ->
         raise (Error ("GtkMain.init: initialization failed\n" ^ err))
     in
+    if setlocale then ignore (Glib.Main.setlocale `NUMERIC (Some "C"));
     Array.blit ~src:argv ~dst:Sys.argv ~len:(Array.length argv)
       ~src_pos:0 ~dst_pos:0;
     Obj.truncate (Obj.repr Sys.argv) (Array.length argv);
